saleor/graphql/discount/mutations/promotion/promotion_create.py

```diff
@@ -208,21 +209,31 @@ def _save_m2m(
         for rule, channels in rules_with_channels_to_add:
             rule.channels.set(channels)
 
+        return rules
+
     @classmethod
-    def send_promotion_webhooks(
-        cls, manager: "PluginsManager", instance: models.Promotion
+    def post_save_actions(
+        cls,
+        manager: "PluginsManager",
+        instance: models.Promotion,
+        info: ResolveInfo,
+        rules: List[models.PromotionRule],
     ):
         cls.call_event(manager.promotion_created, instance)
```

Comment:
Are you sure that you have the correct order of function calls here? 

222 triggers the promotion_created (and webhook if someone is subscribed)
223 optionally triggers `manager.promotion_started` (and probably second webhook).
224 creates new promotion events. It means that the webhooks triggered from 222&223 will not have details from 224